sdk/examples/how_tos/native_tokens/destroy_foundry.rs

```diff
@@ -34,12 +38,50 @@ async fn main() -> Result<()> {
         let foundry_count = balance.foundries().len();
         println!("Foundries before destroying: {foundry_count}");
 
+        let token_id = TokenId::from(*foundry_id);
+
         // Set the stronghold password
         wallet
             .set_stronghold_password(std::env::var("STRONGHOLD_PASSWORD").unwrap())
             .await?;
 
-        let transaction = account.burn(*foundry_id, None).await?;
+        // Check if all tokens are melted.
+        let native_tokens: Option<&NativeTokensBalance> = balance
+            .native_tokens()
+            .iter()
+            .find(|native_token| *native_token.token_id() == token_id);
+        if let Some(native_token) = native_tokens {
+            let output = account.get_foundry_output(token_id).await?;
+            if native_token.available() != output.as_foundry().token_scheme().as_simple().circulating_supply() {
+                // We are not able to melt all tokens, because we dont own them or they are not unlocked.
+                println!("We dont own all remaining tokens, aborting foundry destruction.");
+                return Ok(());
+            }
+
+            println!("Melting remaining tokens..");
+            // Melt all tokens so we can destroy the foundry.
+            let transaction = account
+                .melt_native_token(token_id, native_token.available(), None)
+                .await?;
+            println!("Transaction sent: {}", transaction.transaction_id);
+
+            let block_id = account
+                .retry_transaction_until_included(&transaction.transaction_id, None, None)
+                .await?;
+            println!(
+                "Block included: {}/block/{}",
+                std::env::var("EXPLORER_URL").unwrap(),
+                block_id
+            );
+
+            // Update account so input selection works instead of throwing `UnfulfillableRequirement`.
```

Comment:
Weird comment

Reply:
Yeah I guess it should be rather be something like `Sync to make the foundry output available again, because it was used in the melting transaction`